NewsApp/app/src/main/java/com/yourcompany/newsapp/MainActivity.kt

```diff
@@ -0,0 +1,120 @@
+            "https://www.cnn.com/2022/08/26/health/moderna-pfizer-mrna-patent-lawsuit/index.html",
+            "https://cdn.cnn.com/cnnnext/dam/assets/220815081045-moderna-covid-booster-omicron-variant-super-tease.jpg",
+            "2022-08-26T16:11:00Z",
+            ),
+        Article("Nicolas Vega",
+            "House of the Dragon’ gets renewed for season 2 after setting an HBO record with premiere episode - CNBC",
+            "More than 20 million viewers watched the premiere episode of the prequel series, which is set 170 years before the events of the original show.",
+            "https://www.cnbc.com/2022/08/26/game-of-thrones-hbo-renews-house-of-the-dragon-for-season-2.html",
+            "https://image.cnbcfm.com/api/v1/image/107096109-1659105656073-emma-d-arcy-as-princess-rhaenyra-targaryen-matt-smith-as-prince-daemon-targaryen.jpg?v=1659105811&w=1920&h=1080",
+            "2022-08-26T15:20:40Z",
+            "HBO's \\\"House of the Dragon\\\" hasn't yet aired its second episode, but that hasn't stopped the network from greenlighting it for a second season. \\r\\nThe first episode of the \\\"Game of Thrones\\\" prequel se… [+1401 chars]"),
+        Article(null,
+            "No way Russell Westbrook is going to be WORSE than he was last season - Stephen A. | First Take - ESPN",
+            "On First Take, Stephen A. Smith and Kendrick Perkins debate whether Russell Westbrook needs to be traded, will he have a good season, LeBron's contribution t...",
+            "https://www.youtube.com/watch?v=C9Ipk21VeOU",
+            "https://i.ytimg.com/vi/C9Ipk21VeOU/hqdefault.jpg",
+            "2022-08-26T15:13:55Z",
+         ),
+        Article("https://www.facebook.com/bbcnews",
+            "Pakistan floods: Monsoons bring misery to millions in Pakistan - BBC",
+            "Historic downpours and major floods have battered the nation since June, killing hundreds.",
+            "https://www.bbc.com/news/world-asia-62684037",
+            "https://ichef.bbci.co.uk/news/1024/branded_news/3511/production/_126458531_c3cbed71-c71b-4292-84fc-40c765c73623.jpg",
+            "2022-08-26T15:13:03Z",
+            "By Pumza Fihlani in Sukkur, Pakistan and Frances Mao in SingaporeBBC News\\r\\nMedia caption, Watch: Floods deluge parts of southern Pakistan\\r\\nMillions of people have been affected by floods in Pakistan,… [+2399 chars]"),
+
+
+    )
+
+    private val source = arrayListOf(
+
+        Source("abc-news", "ABC News"),
+        Source("cnn", "CNN"),
+        Source("", "CNBC"),
+        Source("", "YouTube"),
+        Source("", "BBC News")
+    )
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        //setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        array()
+    }
+    private fun array(){
```

Comment:
Nice work making this method `private`!  As a suggestion, you could also provide a name for the function that describes what it does:

```suggestion
    private fun setArticleListText(){
```

Reply:
I totally agree with you suggestion.